### PR TITLE
Implement PouchDbKey

### PR DESCRIPTION
**Pull request by lindner**
Provides an abstraction for db names, locations and three
dbLocations supported by Pouch

- memory
- local (idb/leveldb)
- remote

Addresses: #1946

### CODE REVIEW

**Comment by lindner** [comment]
Starting with simple stuff.

File: runtime/test/storage/pouchdb/pouch-db-key-test.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  });
+
+  it('parses a memory url', () => {
+    let key = 'pouchdb://memory/user/';
+    let pkey = new PouchDbKey(key);
+    expect(pkey.toString()).to.equal(key);
+    expect(pkey.dbLocation).to.equal('memory');
+    expect(pkey.dbName).to.equal('user');
+    expect(pkey.location).to.equal('');
+    expect(pkey.dbCacheKey()).to.equal('memory/user');
```
Comment (mykmartin) [inline runtime/test/storage/pouchdb/pouch-db-key-test.js]: Please use the assert API to be consistent with the rest of the unit tests.

Comment (lindner) [reply runtime/test/storage/pouchdb/pouch-db-key-test.js]: done

**Comment by lindner** [comment]
Regarding key namespaces and behavior, it's all very configurable.

The current implementation assumes:

- Memory keys are volatile and don't get replicated anywhere.
- local database names and remote names have a 1:1 mapping.  The client code interacts with the local version and replicates outwards.  This means that we should be able to operate offline.

This means that if you have

- pouchdb://remotehost1.com/foo
- pouchdb://remotehost2.com/foo

They will both attach to a local 'foo' that contains the data in question, and data will be replicated amongst all three entities through the local client.

**Comment by shans** [comment]
Right, but storageKeys are stored in serialized arcs and intended to be universal. I'm mainly concerned with ensuring that if host A creates a key (whether that be remotehost1.com or whatever) and writes it into a serialized arc, then host B can read that key and immediately use it (without key modification) to read the data that host A wrote through that key.

With regards to memory keys - this seems to be what in-memory stores are already doing.

**Pull request merged**